[
    {
        "number": 291,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-9wx4-h78v-vm56",
            "cve_id": "CVE-2024-35195",
            "summary": "Requests `Session` object does not verify requests after making first request with verify=False",
            "description": "When making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same origin will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool.\n\n### Remediation\nAny of these options can be used to remediate the current issue, we highly recommend upgrading as the preferred mitigation.\n\n* Upgrade to `requests>=2.32.0`.\n* For `requests<2.32.0`, avoid setting `verify=False` for the first request to a host while using a Requests Session.\n* For `requests<2.32.0`, call `close()` on `Session` objects to clear existing connections if `verify=False` is used.\n\n### Related Links\n* https://github.com/psf/requests/pull/6655",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-9wx4-h78v-vm56",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-35195",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56"
                },
                {
                    "url": "https://github.com/psf/requests/pull/6655"
                },
                {
                    "url": "https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-35195"
                },
                {
                    "url": "https://github.com/advisories/GHSA-9wx4-h78v-vm56"
                }
            ],
            "published_at": "2024-05-20T20:15:00Z",
            "updated_at": "2024-05-20T22:07:29Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "requests"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.32.0",
                    "first_patched_version": {
                        "identifier": "2.32.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N",
                "score": 5.6
            },
            "cwes": [
                {
                    "cwe_id": "CWE-670",
                    "name": "Always-Incorrect Control Flow Implementation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.32.0",
            "first_patched_version": {
                "identifier": "2.32.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/291",
        "html_url": "https://github.com/fcas/igel/security/dependabot/291",
        "created_at": "2024-05-24T01:34:22Z",
        "updated_at": "2024-05-24T01:34:22Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 290,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Werkzeug"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-2g68-c3qc-8985",
            "cve_id": "CVE-2024-34069",
            "summary": "Werkzeug debugger vulnerable to remote execution when interacting with attacker controlled domain",
            "description": "The debugger in affected versions of Werkzeug can allow an attacker to execute code on a developer's machine under some circumstances. This requires the attacker to get the developer to interact with a domain and subdomain they control, and enter the debugger PIN, but if they are successful it allows access to the debugger even if it is only running on localhost. This also requires the attacker to guess a URL in the developer's application that will trigger the debugger.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-2g68-c3qc-8985",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-34069",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/werkzeug/security/advisories/GHSA-2g68-c3qc-8985"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/3386395b24c7371db11a5b8eaac0c91da5362692"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34069"
                },
                {
                    "url": "https://github.com/advisories/GHSA-2g68-c3qc-8985"
                }
            ],
            "published_at": "2024-05-06T14:21:27Z",
            "updated_at": "2024-05-06T16:17:35Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Werkzeug"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 3.0.3",
                    "first_patched_version": {
                        "identifier": "3.0.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-352",
                    "name": "Cross-Site Request Forgery (CSRF)"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Werkzeug"
            },
            "severity": "high",
            "vulnerable_version_range": "< 3.0.3",
            "first_patched_version": {
                "identifier": "3.0.3"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/290",
        "html_url": "https://github.com/fcas/igel/security/dependabot/290",
        "created_at": "2024-05-24T01:34:22Z",
        "updated_at": "2024-05-24T01:34:22Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 289,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Jinja2"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h75v-3vvj-5mfj",
            "cve_id": "CVE-2024-34064",
            "summary": "Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter",
            "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for the previous GHSA-h5c8-rqwp-cp95 CVE-2024-22195 only addressed spaces but not other characters.\n\nAccepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-h75v-3vvj-5mfj",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-34064",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj"
                },
                {
                    "url": "https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34064"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h75v-3vvj-5mfj"
                }
            ],
            "published_at": "2024-05-06T14:20:59Z",
            "updated_at": "2024-05-06T16:17:32Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Jinja2"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.1.4",
                    "first_patched_version": {
                        "identifier": "3.1.4"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N",
                "score": 5.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-79",
                    "name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Jinja2"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.1.4",
            "first_patched_version": {
                "identifier": "3.1.4"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/289",
        "html_url": "https://github.com/fcas/igel/security/dependabot/289",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 288,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "tqdm"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-g7vv-2v7x-gj9p",
            "cve_id": "CVE-2024-34062",
            "summary": "tqdm CLI arguments injection attack",
            "description": "### Impact\nAny optional non-boolean CLI arguments (e.g. `--delim`, `--buf-size`, `--manpath`) are passed through python's `eval`, allowing arbitrary code execution. Example:\n\n```sh\npython -m tqdm --manpath=\"\\\" + str(exec(\\\"import os\\nos.system('echo hi && killall python3')\\\")) + \\\"\"\n```\n\n### Patches\nhttps://github.com/tqdm/tqdm/commit/4e613f84ed2ae029559f539464df83fa91feb316 released in `tqdm>=4.66.3`\n\n### Workarounds\nNone\n\n### References\n- https://github.com/tqdm/tqdm/releases/tag/v4.66.3",
            "severity": "low",
            "identifiers": [
                {
                    "value": "GHSA-g7vv-2v7x-gj9p",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-34062",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/tqdm/tqdm/security/advisories/GHSA-g7vv-2v7x-gj9p"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34062"
                },
                {
                    "url": "https://github.com/tqdm/tqdm/commit/4e613f84ed2ae029559f539464df83fa91feb316"
                },
                {
                    "url": "https://github.com/advisories/GHSA-g7vv-2v7x-gj9p"
                }
            ],
            "published_at": "2024-05-03T19:33:28Z",
            "updated_at": "2024-05-03T19:33:29Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "tqdm"
                    },
                    "severity": "low",
                    "vulnerable_version_range": ">= 4.4.0, < 4.66.3",
                    "first_patched_version": {
                        "identifier": "4.66.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:L/UI:R/S:U/C:L/I:L/A:N",
                "score": 3.9
            },
            "cwes": [
                {
                    "cwe_id": "CWE-74",
                    "name": "Improper Neutralization of Special Elements in Output Used by a Downstream Component ('Injection')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "tqdm"
            },
            "severity": "low",
            "vulnerable_version_range": ">= 4.4.0, < 4.66.3",
            "first_patched_version": {
                "identifier": "4.66.3"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/288",
        "html_url": "https://github.com/fcas/igel/security/dependabot/288",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 287,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pydantic"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-mr82-8j83-vxmv",
            "cve_id": "CVE-2024-3772",
            "summary": "Pydantic regular expression denial of service",
            "description": "Regular expression denial of service in Pydantic < 2.4.0, < 1.10.13 allows remote attackers to cause denial of service via a crafted email string.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-mr82-8j83-vxmv",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-3772",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-3772"
                },
                {
                    "url": "https://github.com/pydantic/pydantic/pull/7360"
                },
                {
                    "url": "https://github.com/pydantic/pydantic/commit/59d8f38fd6220e3917c53785dbc70317d6f8e631"
                },
                {
                    "url": "https://github.com/pydantic/pydantic/commit/e4393ae6145c4dadff739990bb0116c6dec3441b"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6JBZLMSH4GAZOVBMT2JUO2LXHY7M2ALI"
                },
                {
                    "url": "https://github.com/advisories/GHSA-mr82-8j83-vxmv"
                }
            ],
            "published_at": "2024-04-15T03:31:00Z",
            "updated_at": "2024-04-26T03:30:29Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pydantic"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.0.0, < 2.4.0",
                    "first_patched_version": {
                        "identifier": "2.4.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pydantic"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 1.10.13",
                    "first_patched_version": {
                        "identifier": "1.10.13"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 5.9
            },
            "cwes": [
                {
                    "cwe_id": "CWE-1333",
                    "name": "Inefficient Regular Expression Complexity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pydantic"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 1.10.13",
            "first_patched_version": {
                "identifier": "1.10.13"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/287",
        "html_url": "https://github.com/fcas/igel/security/dependabot/287",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 286,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "idna"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-jjg7-2v4v-x38h",
            "cve_id": "CVE-2024-3651",
            "summary": "Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode",
            "description": "### Impact\nA specially crafted argument to the `idna.encode()` function could consume significant resources. This may lead to a denial-of-service.\n\n### Patches\nThe function has been refined to reject such strings without the associated resource consumption in version 3.7.\n\n### Workarounds\nDomain names cannot exceed 253 characters in length, if this length limit is enforced prior to passing the domain to the `idna.encode()` function it should no longer consume significant resources. This is triggered by arbitrarily large inputs that would not occur in normal usage, but may be passed to the library assuming there is no preliminary input validation by the higher-level application.\n\n### References\n* https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-jjg7-2v4v-x38h",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-3651",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/kjd/idna/security/advisories/GHSA-jjg7-2v4v-x38h"
                },
                {
                    "url": "https://github.com/advisories/GHSA-jjg7-2v4v-x38h"
                }
            ],
            "published_at": "2024-04-11T21:32:40Z",
            "updated_at": "2024-04-11T21:32:40Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "idna"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.7",
                    "first_patched_version": {
                        "identifier": "3.7"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 6.2
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "idna"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.7",
            "first_patched_version": {
                "identifier": "3.7"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/286",
        "html_url": "https://github.com/fcas/igel/security/dependabot/286",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 285,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-44wm-f244-xhp3",
            "cve_id": "CVE-2024-28219",
            "summary": "Pillow buffer overflow vulnerability",
            "description": "In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-44wm-f244-xhp3",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-28219",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-28219"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/2a93aba5cfcf6e241ab4f9392c13e3b74032c061"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M"
                },
                {
                    "url": "https://github.com/advisories/GHSA-44wm-f244-xhp3"
                }
            ],
            "published_at": "2024-04-03T03:30:30Z",
            "updated_at": "2024-04-20T05:06:18Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 10.3.0",
                    "first_patched_version": {
                        "identifier": "10.3.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H",
                "score": 6.7
            },
            "cwes": [
                {
                    "cwe_id": "CWE-120",
                    "name": "Buffer Copy without Checking Size of Input ('Classic Buffer Overflow')"
                },
                {
                    "cwe_id": "CWE-676",
                    "name": "Use of Potentially Dangerous Function"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 10.3.0",
            "first_patched_version": {
                "identifier": "10.3.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/285",
        "html_url": "https://github.com/fcas/igel/security/dependabot/285",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 284,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "black"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-fj7x-q9j7-g6q6",
            "cve_id": "CVE-2024-21503",
            "summary": "Black vulnerable to Regular Expression Denial of Service (ReDoS)",
            "description": "Versions of the package black before 24.3.0 are vulnerable to Regular Expression Denial of Service (ReDoS) via the lines_with_leading_tabs_expanded function in the strings.py file. An attacker could exploit this vulnerability by crafting a malicious input that causes a denial of service.\n\nExploiting this vulnerability is possible when running Black on untrusted input, or if you habitually put thousands of leading tab characters in your docstrings.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-fj7x-q9j7-g6q6",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-21503",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-21503"
                },
                {
                    "url": "https://github.com/psf/black/commit/f00093672628d212b8965a8993cee8bedf5fe9b8"
                },
                {
                    "url": "https://github.com/psf/black/releases/tag/24.3.0"
                },
                {
                    "url": "https://security.snyk.io/vuln/SNYK-PYTHON-BLACK-6256273"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/black/PYSEC-2024-48.yaml"
                },
                {
                    "url": "https://github.com/advisories/GHSA-fj7x-q9j7-g6q6"
                }
            ],
            "published_at": "2024-03-19T06:30:52Z",
            "updated_at": "2024-03-20T15:24:02Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "black"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 0, < 24.3.0",
                    "first_patched_version": {
                        "identifier": "24.3.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L",
                "score": 5.3
            },
            "cwes": [
                {
                    "cwe_id": "CWE-1333",
                    "name": "Inefficient Regular Expression Complexity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "black"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 0, < 24.3.0",
            "first_patched_version": {
                "identifier": "24.3.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/284",
        "html_url": "https://github.com/fcas/igel/security/dependabot/284",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 283,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "onnx"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h8wv-9h96-m4hr",
            "cve_id": "CVE-2024-27319",
            "summary": "Onnx Out-of-bounds Read vulnerability",
            "description": "Versions of the package onnx before and including 1.15.0 are vulnerable to Out-of-bounds Read as the ONNX_ASSERT and ONNX_ASSERTM functions have an off by one string copy.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-h8wv-9h96-m4hr",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-27319",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-27319"
                },
                {
                    "url": "https://github.com/onnx/onnx/commit/08a399ba75a805b7813ab8936b91d0e274b08287"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FGTBH5ZYL2LGYHIJDHN2MAUURIR5E7PY"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TFJJID2IZDOLFDMWVYTBDI75ZJQC6JOL"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h8wv-9h96-m4hr"
                }
            ],
            "published_at": "2024-02-23T18:30:59Z",
            "updated_at": "2024-04-24T14:38:10Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "onnx"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "<= 1.15.0",
                    "first_patched_version": {
                        "identifier": "1.16.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:L/I:N/A:L",
                "score": 4.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-125",
                    "name": "Out-of-bounds Read"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "onnx"
            },
            "severity": "medium",
            "vulnerable_version_range": "<= 1.15.0",
            "first_patched_version": {
                "identifier": "1.16.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/283",
        "html_url": "https://github.com/fcas/igel/security/dependabot/283",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 282,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "onnx"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-whh8-fjgc-qp73",
            "cve_id": "CVE-2024-27318",
            "summary": "Onnx Directory Traversal vulnerability",
            "description": "Versions of the package onnx before and including 1.15.0 are vulnerable to Directory Traversal as the external_data field of the tensor proto can have a path to the file which is outside the model current directory or user-provided directory. The vulnerability occurs as a bypass for the patch added for CVE-2022-25882.\n",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-whh8-fjgc-qp73",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-27318",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-27318"
                },
                {
                    "url": "https://github.com/onnx/onnx/commit/66b7fb630903fdcf3e83b6b6d56d82e904264a20"
                },
                {
                    "url": "https://security.snyk.io/vuln/SNYK-PYTHON-ONNX-2395479"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FGTBH5ZYL2LGYHIJDHN2MAUURIR5E7PY"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TFJJID2IZDOLFDMWVYTBDI75ZJQC6JOL"
                },
                {
                    "url": "https://github.com/advisories/GHSA-whh8-fjgc-qp73"
                }
            ],
            "published_at": "2024-02-23T18:30:59Z",
            "updated_at": "2024-04-24T14:35:49Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "onnx"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 1.15.0",
                    "first_patched_version": {
                        "identifier": "1.16.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-22",
                    "name": "Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "onnx"
            },
            "severity": "high",
            "vulnerable_version_range": "<= 1.15.0",
            "first_patched_version": {
                "identifier": "1.16.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/282",
        "html_url": "https://github.com/fcas/igel/security/dependabot/282",
        "created_at": "2024-05-24T01:34:21Z",
        "updated_at": "2024-05-24T01:34:21Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 281,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "starlette"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-2jv5-9r88-3w3p",
            "cve_id": "CVE-2024-24762",
            "summary": "python-multipart vulnerable to Content-Type Header ReDoS",
            "description": "### Summary\n\nWhen using form data, `python-multipart` uses a Regular Expression to parse the HTTP `Content-Type` header, including options.\n\nAn attacker could send a custom-made `Content-Type` option that is very difficult for the RegEx to process, consuming CPU resources and stalling indefinitely (minutes or more) while holding the main event loop. This means that process can't handle any more requests.\n\nThis can create a ReDoS (Regular expression Denial of Service): https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS\n\nThis only applies when the app uses form data, parsed with `python-multipart`.\n\n### Details\n\nA regular HTTP `Content-Type` header could look like:\n\n```\nContent-Type: text/html; charset=utf-8\n```\n\n`python-multipart` parses the option with this RegEx: https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74\n\nA custom option could be made and sent to the server to break it with:\n\n```\nContent-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\n```\n\n### PoC\n\nCreate a simple WSGI application, that just parses the `Content-Type`, and run it with `python main.py`:\n\n```Python\n# main.py\nfrom wsgiref.simple_server import make_server\nfrom wsgiref.validate import validator\n\nfrom multipart.multipart import parse_options_header\n\n\ndef simple_app(environ, start_response):\n    _, _ = parse_options_header(environ[\"CONTENT_TYPE\"])\n\n    start_response(\"200 OK\", [(\"Content-type\", \"text/plain\")])\n    return [b\"Ok\"]\n\n\nhttpd = make_server(\"\", 8123, validator(simple_app))\nprint(\"Serving on port 8123...\")\nhttpd.serve_forever()\n```\n\nThen send the attacking request with:\n\n```console\n$ curl -v -X 'POST' -H $'Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\' --data-binary 'input=1' 'http://localhost:8123/'\n```\n\n### Impact\n\nIt's a ReDoS, (Regular expression Denial of Service), it only applies to those reading form data. This way it also affects other libraries using Starlette, like FastAPI.\n\n### Original Report\n\nThis was originally reported to FastAPI as an email to security@tiangolo.com, sent via https://huntr.com/, the original reporter is Marcello, https://github.com/byt3bl33d3r\n\n<details>\n<summary>Original report to FastAPI</summary>\n\nHey Tiangolo!\n\nMy name's Marcello and I work on the ProtectAI/Huntr Threat Research team, a few months ago we got a report (from @nicecatch2000) of a ReDoS affecting another very popular Python web framework. After some internal research, I found that FastAPI is vulnerable to the same ReDoS under certain conditions (only when it parses Form data not JSON).\n\nHere are the details: I'm using the latest version of FastAPI (0.109.0) and the following code:\n\n```Python\nfrom typing import Annotated\nfrom fastapi.responses import HTMLResponse\nfrom fastapi import FastAPI,Form\nfrom pydantic import BaseModel\n\nclass Item(BaseModel):\n    username: str\n\napp = FastAPI()\n\n@app.get(\"/\", response_class=HTMLResponse)\nasync def index():\n    return HTMLResponse(\"Test\", status_code=200)\n\n@app.post(\"/submit/\")\nasync def submit(username: Annotated[str, Form()]):\n    return {\"username\": username}\n\n@app.post(\"/submit_json/\")\nasync def submit_json(item: Item):\n    return {\"username\": item.username}\n```\n\nI'm running the above with uvicorn with the following command:\n\n```console\nuvicorn server:app\n```\n\nThen run the following cUrl command:\n\n```\ncurl -v -X 'POST' -H $'Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\' --data-binary 'input=1' 'http://localhost:8000/submit/'\n```\n\nYou'll see the server locks up, is unable to serve anymore requests and one CPU core is pegged to 100%\n\nYou can even start uvicorn with multiple workers with the --workers 4 argument and as long as you send (workers + 1) requests you'll completely DoS the FastApi server.\n\nIf you try submitting Json to the /submit_json endpoint with the malicious Content-Type header you'll see it isn't vulnerable. So this only affects FastAPI when it parses Form data.\n\nCheers\n\n#### Impact\n\nAn attacker is able to cause a DoS on a FastApi server via a malicious Content-Type header if it parses Form data.\n\n#### Occurrences\n\n[params.py L586](https://github.com/tiangolo/fastapi/blob/d74b3b25659b42233a669f032529880de8bd6c2d/fastapi/params.py#L586)\n\n</details>",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-2jv5-9r88-3w3p",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-24762",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/Kludex/python-multipart/security/advisories/GHSA-2jv5-9r88-3w3p"
                },
                {
                    "url": "https://github.com/Kludex/python-multipart/commit/20f0ef6b4e4caf7d69a667c54dff57fe467109a4"
                },
                {
                    "url": "https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74"
                },
                {
                    "url": "https://github.com/encode/starlette/security/advisories/GHSA-93gm-qmq6-w238"
                },
                {
                    "url": "https://github.com/tiangolo/fastapi/security/advisories/GHSA-qf9m-vfgh-m389"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-24762"
                },
                {
                    "url": "https://github.com/encode/starlette/commit/13e5c26a27f4903924624736abd6131b2da80cc5"
                },
                {
                    "url": "https://github.com/tiangolo/fastapi/commit/9d34ad0ee8a0dfbbcce06f76c2d5d851085024fc"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/fastapi/PYSEC-2024-38.yaml"
                },
                {
                    "url": "https://github.com/tiangolo/fastapi/releases/tag/0.109.1"
                },
                {
                    "url": "https://github.com/advisories/GHSA-2jv5-9r88-3w3p"
                }
            ],
            "published_at": "2024-02-12T17:28:12Z",
            "updated_at": "2024-02-16T23:41:49Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "python-multipart"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.0.6",
                    "first_patched_version": {
                        "identifier": "0.0.7"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "fastapi"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.109.0",
                    "first_patched_version": {
                        "identifier": "0.109.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "starlette"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.36.1",
                    "first_patched_version": {
                        "identifier": "0.36.2"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "starlette"
            },
            "severity": "high",
            "vulnerable_version_range": "<= 0.36.1",
            "first_patched_version": {
                "identifier": "0.36.2"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/281",
        "html_url": "https://github.com/fcas/igel/security/dependabot/281",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 280,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "fastapi"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-2jv5-9r88-3w3p",
            "cve_id": "CVE-2024-24762",
            "summary": "python-multipart vulnerable to Content-Type Header ReDoS",
            "description": "### Summary\n\nWhen using form data, `python-multipart` uses a Regular Expression to parse the HTTP `Content-Type` header, including options.\n\nAn attacker could send a custom-made `Content-Type` option that is very difficult for the RegEx to process, consuming CPU resources and stalling indefinitely (minutes or more) while holding the main event loop. This means that process can't handle any more requests.\n\nThis can create a ReDoS (Regular expression Denial of Service): https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS\n\nThis only applies when the app uses form data, parsed with `python-multipart`.\n\n### Details\n\nA regular HTTP `Content-Type` header could look like:\n\n```\nContent-Type: text/html; charset=utf-8\n```\n\n`python-multipart` parses the option with this RegEx: https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74\n\nA custom option could be made and sent to the server to break it with:\n\n```\nContent-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\n```\n\n### PoC\n\nCreate a simple WSGI application, that just parses the `Content-Type`, and run it with `python main.py`:\n\n```Python\n# main.py\nfrom wsgiref.simple_server import make_server\nfrom wsgiref.validate import validator\n\nfrom multipart.multipart import parse_options_header\n\n\ndef simple_app(environ, start_response):\n    _, _ = parse_options_header(environ[\"CONTENT_TYPE\"])\n\n    start_response(\"200 OK\", [(\"Content-type\", \"text/plain\")])\n    return [b\"Ok\"]\n\n\nhttpd = make_server(\"\", 8123, validator(simple_app))\nprint(\"Serving on port 8123...\")\nhttpd.serve_forever()\n```\n\nThen send the attacking request with:\n\n```console\n$ curl -v -X 'POST' -H $'Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\' --data-binary 'input=1' 'http://localhost:8123/'\n```\n\n### Impact\n\nIt's a ReDoS, (Regular expression Denial of Service), it only applies to those reading form data. This way it also affects other libraries using Starlette, like FastAPI.\n\n### Original Report\n\nThis was originally reported to FastAPI as an email to security@tiangolo.com, sent via https://huntr.com/, the original reporter is Marcello, https://github.com/byt3bl33d3r\n\n<details>\n<summary>Original report to FastAPI</summary>\n\nHey Tiangolo!\n\nMy name's Marcello and I work on the ProtectAI/Huntr Threat Research team, a few months ago we got a report (from @nicecatch2000) of a ReDoS affecting another very popular Python web framework. After some internal research, I found that FastAPI is vulnerable to the same ReDoS under certain conditions (only when it parses Form data not JSON).\n\nHere are the details: I'm using the latest version of FastAPI (0.109.0) and the following code:\n\n```Python\nfrom typing import Annotated\nfrom fastapi.responses import HTMLResponse\nfrom fastapi import FastAPI,Form\nfrom pydantic import BaseModel\n\nclass Item(BaseModel):\n    username: str\n\napp = FastAPI()\n\n@app.get(\"/\", response_class=HTMLResponse)\nasync def index():\n    return HTMLResponse(\"Test\", status_code=200)\n\n@app.post(\"/submit/\")\nasync def submit(username: Annotated[str, Form()]):\n    return {\"username\": username}\n\n@app.post(\"/submit_json/\")\nasync def submit_json(item: Item):\n    return {\"username\": item.username}\n```\n\nI'm running the above with uvicorn with the following command:\n\n```console\nuvicorn server:app\n```\n\nThen run the following cUrl command:\n\n```\ncurl -v -X 'POST' -H $'Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\' --data-binary 'input=1' 'http://localhost:8000/submit/'\n```\n\nYou'll see the server locks up, is unable to serve anymore requests and one CPU core is pegged to 100%\n\nYou can even start uvicorn with multiple workers with the --workers 4 argument and as long as you send (workers + 1) requests you'll completely DoS the FastApi server.\n\nIf you try submitting Json to the /submit_json endpoint with the malicious Content-Type header you'll see it isn't vulnerable. So this only affects FastAPI when it parses Form data.\n\nCheers\n\n#### Impact\n\nAn attacker is able to cause a DoS on a FastApi server via a malicious Content-Type header if it parses Form data.\n\n#### Occurrences\n\n[params.py L586](https://github.com/tiangolo/fastapi/blob/d74b3b25659b42233a669f032529880de8bd6c2d/fastapi/params.py#L586)\n\n</details>",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-2jv5-9r88-3w3p",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-24762",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/Kludex/python-multipart/security/advisories/GHSA-2jv5-9r88-3w3p"
                },
                {
                    "url": "https://github.com/Kludex/python-multipart/commit/20f0ef6b4e4caf7d69a667c54dff57fe467109a4"
                },
                {
                    "url": "https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74"
                },
                {
                    "url": "https://github.com/encode/starlette/security/advisories/GHSA-93gm-qmq6-w238"
                },
                {
                    "url": "https://github.com/tiangolo/fastapi/security/advisories/GHSA-qf9m-vfgh-m389"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-24762"
                },
                {
                    "url": "https://github.com/encode/starlette/commit/13e5c26a27f4903924624736abd6131b2da80cc5"
                },
                {
                    "url": "https://github.com/tiangolo/fastapi/commit/9d34ad0ee8a0dfbbcce06f76c2d5d851085024fc"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/fastapi/PYSEC-2024-38.yaml"
                },
                {
                    "url": "https://github.com/tiangolo/fastapi/releases/tag/0.109.1"
                },
                {
                    "url": "https://github.com/advisories/GHSA-2jv5-9r88-3w3p"
                }
            ],
            "published_at": "2024-02-12T17:28:12Z",
            "updated_at": "2024-02-16T23:41:49Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "python-multipart"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.0.6",
                    "first_patched_version": {
                        "identifier": "0.0.7"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "fastapi"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.109.0",
                    "first_patched_version": {
                        "identifier": "0.109.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "starlette"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 0.36.1",
                    "first_patched_version": {
                        "identifier": "0.36.2"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "fastapi"
            },
            "severity": "high",
            "vulnerable_version_range": "<= 0.109.0",
            "first_patched_version": {
                "identifier": "0.109.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/280",
        "html_url": "https://github.com/fcas/igel/security/dependabot/280",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 279,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-9v9h-cgj8-h64p",
            "cve_id": "CVE-2024-0727",
            "summary": "Null pointer dereference in PKCS12 parsing",
            "description": "Issue summary: Processing a maliciously formatted PKCS12 file may lead OpenSSL\nto crash leading to a potential Denial of Service attack\n\nImpact summary: Applications loading files in the PKCS12 format from untrusted\nsources might terminate abruptly.\n\nA file in PKCS12 format can contain certificates and keys and may come from an\nuntrusted source. The PKCS12 specification allows certain fields to be NULL, but\nOpenSSL does not correctly check for this case. This can lead to a NULL pointer\ndereference that results in OpenSSL crashing. If an application processes PKCS12\nfiles from an untrusted source using the OpenSSL APIs then that application will\nbe vulnerable to this issue.\n\nOpenSSL APIs that are vulnerable to this are: PKCS12_parse(),\nPKCS12_unpack_p7data(), PKCS12_unpack_p7encdata(), PKCS12_unpack_authsafes()\nand PKCS12_newpass().\n\nWe have also fixed a similar issue in SMIME_write_PKCS7(). However since this\nfunction is related to writing data we do not consider it security significant.\n\nThe FIPS modules in 3.2, 3.1 and 3.0 are not affected by this issue.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-9v9h-cgj8-h64p",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-0727",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-0727"
                },
                {
                    "url": "https://github.com/openssl/openssl/commit/09df4395b5071217b76dc7d3d2e630eb8c5a79c2"
                },
                {
                    "url": "https://github.com/openssl/openssl/commit/775acfdbd0c6af9ac855f34969cdab0c0c90844a"
                },
                {
                    "url": "https://github.com/openssl/openssl/commit/d135eeab8a5dbf72b3da5240bab9ddb7678dbd2c"
                },
                {
                    "url": "https://github.openssl.org/openssl/extended-releases/commit/03b3941d60c4bce58fab69a0c22377ab439bc0e8"
                },
                {
                    "url": "https://github.openssl.org/openssl/extended-releases/commit/aebaa5883e31122b404e450732dc833dc9dee539"
                },
                {
                    "url": "https://www.openssl.org/news/secadv/20240125.txt"
                },
                {
                    "url": "https://github.com/github/advisory-database/pull/3472"
                },
                {
                    "url": "https://github.com/openssl/openssl/pull/23362"
                },
                {
                    "url": "https://github.com/alexcrichton/openssl-src-rs/commit/add20f73b6b42be7451af2e1044d4e0e778992b2"
                },
                {
                    "url": "https://github.com/pyca/cryptography/commit/3519591d255d4506fbcd0d04037d45271903c64d"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20240208-0006"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2024/03/11/1"
                },
                {
                    "url": "https://github.com/advisories/GHSA-9v9h-cgj8-h64p"
                }
            ],
            "published_at": "2024-01-26T09:30:23Z",
            "updated_at": "2024-05-12T05:03:47Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "cryptography"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 42.0.2",
                    "first_patched_version": {
                        "identifier": "42.0.2"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H",
                "score": 5.5
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 42.0.2",
            "first_patched_version": {
                "identifier": "42.0.2"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/279",
        "html_url": "https://github.com/fcas/igel/security/dependabot/279",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 278,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-3ww4-gg4f-jr7f",
            "cve_id": "CVE-2023-50782",
            "summary": "Python Cryptography package vulnerable to Bleichenbacher timing oracle attack",
            "description": "A flaw was found in the python-cryptography package. This issue may allow a remote attacker to decrypt captured messages in TLS servers that use RSA key exchanges, which may lead to exposure of confidential or sensitive data.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-3ww4-gg4f-jr7f",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-50782",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-50782"
                },
                {
                    "url": "https://access.redhat.com/security/cve/CVE-2023-50782"
                },
                {
                    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2254432"
                },
                {
                    "url": "https://github.com/pyca/cryptography/issues/9785"
                },
                {
                    "url": "https://github.com/advisories/GHSA-3ww4-gg4f-jr7f"
                }
            ],
            "published_at": "2024-02-05T21:30:31Z",
            "updated_at": "2024-02-14T21:31:39Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "cryptography"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 42.0.0",
                    "first_patched_version": {
                        "identifier": "42.0.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-203",
                    "name": "Observable Discrepancy"
                },
                {
                    "cwe_id": "CWE-208",
                    "name": "Observable Timing Discrepancy"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "severity": "high",
            "vulnerable_version_range": "< 42.0.0",
            "first_patched_version": {
                "identifier": "42.0.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/278",
        "html_url": "https://github.com/fcas/igel/security/dependabot/278",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 277,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "scikit-learn"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-jxfp-4rvq-9h9m",
            "cve_id": "CVE-2020-28975",
            "summary": "scikit-learn Denial of Service",
            "description": "svm_predict_values in svm.cpp in Libsvm v324, as used in scikit-learn 0.23.2 and other products, allows attackers to cause a denial of service (segmentation fault) via a crafted model SVM (introduced via pickle, json, or any other model permanence standard) with a large value in the _n_support array.\nNOTE: the scikit-learn vendor's position is that the behavior can only occur if the library's API is violated by an application that changes a private attribute.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-jxfp-4rvq-9h9m",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2020-28975",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2020-28975"
                },
                {
                    "url": "https://github.com/scikit-learn/scikit-learn/issues/18891"
                },
                {
                    "url": "https://github.com/cjlin1/libsvm/blob/9a3a9708926dec87d382c43b203f2ca19c2d56a0/svm.cpp#L2501"
                },
                {
                    "url": "http://packetstormsecurity.com/files/160281/SciKit-Learn-0.23.2-Denial-Of-Service.html"
                },
                {
                    "url": "http://seclists.org/fulldisclosure/2020/Nov/44"
                },
                {
                    "url": "https://github.com/scikit-learn/scikit-learn/commit/1bf13d567d3cd74854aa8343fd25b61dd768bb85"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202301-03"
                },
                {
                    "url": "https://github.com/advisories/GHSA-jxfp-4rvq-9h9m"
                }
            ],
            "published_at": "2022-05-24T17:34:40Z",
            "updated_at": "2024-02-01T20:59:17Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "scikit-learn"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 0.23.2, < 1.0.1",
                    "first_patched_version": {
                        "identifier": "1.0.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "scikit-learn"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 0.23.2, < 1.0.1",
            "first_patched_version": {
                "identifier": "1.0.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/277",
        "html_url": "https://github.com/fcas/igel/security/dependabot/277",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 276,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-3f63-hfp8-52jq",
            "cve_id": "CVE-2023-50447",
            "summary": "Arbitrary Code Execution in Pillow",
            "description": "Pillow through 10.1.0 allows PIL.ImageMath.eval Arbitrary Code Execution via the environment parameter, a different vulnerability than CVE-2022-22817 (which was about the expression parameter).",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-3f63-hfp8-52jq",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-50447",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-50447"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/45c726fd4daa63236a8f3653530f297dc87b160a"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/10.2.0.html#security"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2024/01/20/1"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/releases"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html"
                },
                {
                    "url": "https://devhub.checkmarx.com/cve-details/CVE-2023-50447"
                },
                {
                    "url": "https://duartecsantos.github.io/2023-01-02-CVE-2023-50447"
                },
                {
                    "url": "https://duartecsantos.github.io/2024-01-02-CVE-2023-50447"
                },
                {
                    "url": "https://github.com/advisories/GHSA-3f63-hfp8-52jq"
                }
            ],
            "published_at": "2024-01-19T21:30:35Z",
            "updated_at": "2024-03-27T23:53:02Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 10.2.0",
                    "first_patched_version": {
                        "identifier": "10.2.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
                "score": 8.1
            },
            "cwes": [
                {
                    "cwe_id": "CWE-94",
                    "name": "Improper Control of Generation of Code ('Code Injection')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 10.2.0",
            "first_patched_version": {
                "identifier": "10.2.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/276",
        "html_url": "https://github.com/fcas/igel/security/dependabot/276",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 275,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "jinja2"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h5c8-rqwp-cp95",
            "cve_id": "CVE-2024-22195",
            "summary": "Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter",
            "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing spaces. XML/HTML attributes cannot contain spaces, as each would then be interpreted as a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. Note that accepting keys as user input is not common or a particularly intended use case of the `xmlattr` filter, and an application doing so should already be verifying what keys are provided regardless of this fix.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-h5c8-rqwp-cp95",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-22195",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/jinja/security/advisories/GHSA-h5c8-rqwp-cp95"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-22195"
                },
                {
                    "url": "https://github.com/pallets/jinja/commit/716795349a41d4983a9a4771f7d883c96ea17be7"
                },
                {
                    "url": "https://github.com/pallets/jinja/releases/tag/3.1.3"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5XCWZD464AJJJUBOO7CMPXQ4ROBC6JX2/"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/01/msg00010.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DELCVUUYX75I5K4Q5WMJG4MUZJA6VAIP/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O7YWRBX6JQCWC2XXCTZ55C7DPMGICCN3/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h5c8-rqwp-cp95"
                }
            ],
            "published_at": "2024-01-11T15:20:48Z",
            "updated_at": "2024-01-27T03:30:20Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "jinja2"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.1.3",
                    "first_patched_version": {
                        "identifier": "3.1.3"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N",
                "score": 5.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-79",
                    "name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "jinja2"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.1.3",
            "first_patched_version": {
                "identifier": "3.1.3"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/275",
        "html_url": "https://github.com/fcas/igel/security/dependabot/275",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 274,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "GitPython"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-2mqj-m65w-jghx",
            "cve_id": "CVE-2024-22190",
            "summary": "Untrusted search path under some conditions on Windows allows arbitrary code execution",
            "description": "### Summary\n\nThis issue exists because of an incomplete fix for CVE-2023-40590. On Windows, GitPython uses an untrusted search path if it uses a shell to run `git`, as well as when it runs `bash.exe` to interpret hooks. If either of those features are used on Windows, a malicious `git.exe` or `bash.exe` may be run from an untrusted repository.\n\n### Details\n\nAlthough GitPython often avoids executing programs found in an untrusted search path since 3.1.33, two situations remain where this still occurs. Either can allow arbitrary code execution under some circumstances.\n\n#### When a shell is used\n\nGitPython can be told to run `git` commands through a shell rather than as direct subprocesses, by passing `shell=True` to any method that accepts it, or by both setting `Git.USE_SHELL = True` and not passing `shell=False`. Then the Windows `cmd.exe` shell process performs the path search, and GitPython does not prevent that shell from finding and running `git` in the current directory.\n\nWhen GitPython runs `git` directly rather than through a shell, the GitPython process performs the path search, and currently omits the current directory by setting `NoDefaultCurrentDirectoryInExePath` in its own environment during the `Popen` call. Although the `cmd.exe` shell will honor this environment variable when present, GitPython does not currently pass it into the shell subprocess's environment.\n\nFurthermore, because GitPython sets the subprocess CWD to the root of a repository's working tree, using a shell will run a malicious `git.exe` in an untrusted repository even if GitPython itself is run from a trusted location.\n\nThis also applies if `Git.execute` is called directly with `shell=True` (or after `Git.USE_SHELL = True`) to run any command.\n\n#### When hook scripts are run\n\nOn Windows, GitPython uses `bash.exe` to run hooks that appear to be scripts. However, unlike when running `git`, no steps are taken to avoid finding and running `bash.exe` in the current directory.\n\nThis allows the author of an untrusted fork or branch to cause a malicious `bash.exe` to be run in some otherwise safe workflows. An example of such a scenario is if the user installs a trusted hook while on a trusted branch, then switches to an untrusted feature branch (possibly from a fork) to review proposed changes. If the untrusted feature branch contains a malicious `bash.exe` and the user's current working directory is the working tree, and the user performs an action that runs the hook, then although the hook itself is uncorrupted, it runs with the malicious `bash.exe`.\n\nNote that, while `bash.exe` is a shell, this is a separate scenario from when `git` is run using the unrelated Windows `cmd.exe` shell.\n\n### PoC\n\nOn Windows, create a `git.exe` file in a repository. Then create a `Repo` object, and call any method through it (directly or indirectly) that supports the `shell` keyword argument with `shell=True`:\n\n```powershell\nmkdir testrepo\ngit init testrepo\ncp ... testrepo git.exe # Replace \"...\" with any executable of choice.\npython -c \"import git; print(git.Repo('testrepo').git.version(shell=True))\"\n```\n\nThe `git.exe` executable in the repository directory will be run.\n\nOr use no `Repo` object, but do it from the location with the `git.exe`:\n\n```powershell\ncd testrepo\npython -c \"import git; print(git.Git().version(shell=True))\"\n```\n\nThe `git.exe` executable in the current directory will be run.\n\nFor the scenario with hooks, install a hook in a repository, create a `bash.exe` file in the current directory, and perform an operation that causes GitPython to attempt to run the hook:\n\n```powershell\nmkdir testrepo\ncd testrepo\ngit init\nmv .git/hooks/pre-commit.sample .git/hooks/pre-commit\ncp ... bash.exe # Replace \"...\" with any executable of choice.\necho \"Some text\" >file.txt\ngit add file.txt\npython -c \"import git; git.Repo().index.commit('Some message')\"\n```\n\nThe `bash.exe` executable in the current directory will be run.\n\n### Impact\n\nThe greatest impact is probably in applications that set `Git.USE_SHELL = True` for historical reasons. (Undesired console windows had, in the past, been created in some kinds of applications, when it was not used.) Such an application may be vulnerable to arbitrary code execution from a malicious repository, even with no other exacerbating conditions. This is to say that, if a shell is used to run `git`, the full effect of CVE-2023-40590 is still present. Furthermore, as noted above, running the application itself from a trusted directory is not a sufficient mitigation.\n\nAn application that does not direct GitPython to use a shell to run `git` subprocesses thus avoids most of the risk. However, there is no such straightforward way to prevent GitPython from running `bash.exe` to interpret hooks. So while the conditions needed for that to be exploited are more involved, it may be harder to mitigate decisively prior to patching.\n\n### Possible solutions\n\nA straightforward approach would be to address each bug directly:\n\n- When a shell is used, pass `NoDefaultCurrentDirectoryInExePath` into the subprocess environment, because in that scenario the subprocess is the `cmd.exe` shell that itself performs the path search.\n- Set `NoDefaultCurrentDirectoryInExePath` in the GitPython process environment during the `Popen` call made to run hooks with a `bash.exe` subprocess.\n\nThese need only be done on Windows.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-2mqj-m65w-jghx",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-22190",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-2mqj-m65w-jghx"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/pull/1792"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/commit/ef3192cc414f2fd9978908454f6fd95243784c7f"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-22190"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/gitpython/PYSEC-2024-4.yaml"
                },
                {
                    "url": "https://github.com/advisories/GHSA-2mqj-m65w-jghx"
                }
            ],
            "published_at": "2024-01-10T15:46:00Z",
            "updated_at": "2024-01-19T16:24:51Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "GitPython"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 3.1.41",
                    "first_patched_version": {
                        "identifier": "3.1.41"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "score": 7.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-426",
                    "name": "Untrusted Search Path"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "GitPython"
            },
            "severity": "high",
            "vulnerable_version_range": "< 3.1.41",
            "first_patched_version": {
                "identifier": "3.1.41"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/274",
        "html_url": "https://github.com/fcas/igel/security/dependabot/274",
        "created_at": "2024-05-24T01:34:20Z",
        "updated_at": "2024-05-24T01:34:20Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 273,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-jfhm-5ghh-2f97",
            "cve_id": "CVE-2023-49083",
            "summary": "cryptography vulnerable to NULL-dereference when loading PKCS7 certificates",
            "description": "### Summary\n\nCalling `load_pem_pkcs7_certificates` or `load_der_pkcs7_certificates` could lead to a NULL-pointer dereference and segfault.\n\n### PoC\nHere is a Python code that triggers the issue:\n```python\nfrom cryptography.hazmat.primitives.serialization.pkcs7 import load_der_pkcs7_certificates, load_pem_pkcs7_certificates\n\npem_p7 = b\"\"\"\n-----BEGIN PKCS7-----\nMAsGCSqGSIb3DQEHAg==\n-----END PKCS7-----\n\"\"\"\n\nder_p7 = b\"\\x30\\x0B\\x06\\x09\\x2A\\x86\\x48\\x86\\xF7\\x0D\\x01\\x07\\x02\"\n\nload_pem_pkcs7_certificates(pem_p7)\nload_der_pkcs7_certificates(der_p7)\n```\n\n### Impact\nExploitation of this vulnerability poses a serious risk of Denial of Service (DoS) for any application attempting to deserialize a PKCS7 blob/certificate. The consequences extend to potential disruptions in system availability and stability.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-jfhm-5ghh-2f97",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-49083",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pyca/cryptography/security/advisories/GHSA-jfhm-5ghh-2f97"
                },
                {
                    "url": "https://github.com/pyca/cryptography/commit/f09c261ca10a31fe41b1262306db7f8f1da0e48a"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-49083"
                },
                {
                    "url": "https://github.com/pyca/cryptography/pull/9926"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/11/29/2"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/cryptography/PYSEC-2023-254.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QMNTYMUGFJSDBYBU22FUYBHFRZODRKXV"
                },
                {
                    "url": "https://github.com/advisories/GHSA-jfhm-5ghh-2f97"
                }
            ],
            "published_at": "2023-11-28T20:46:46Z",
            "updated_at": "2024-02-20T18:14:36Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "cryptography"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 3.1, < 41.0.6",
                    "first_patched_version": {
                        "identifier": "41.0.6"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 5.9
            },
            "cwes": [
                {
                    "cwe_id": "CWE-476",
                    "name": "NULL Pointer Dereference"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 3.1, < 41.0.6",
            "first_patched_version": {
                "identifier": "41.0.6"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/273",
        "html_url": "https://github.com/fcas/igel/security/dependabot/273",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 272,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "werkzeug"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-hrfv-mqp8-q5rw",
            "cve_id": "CVE-2023-46136",
            "summary": "Werkzeug DoS: High resource usage when parsing multipart/form-data containing a large part with CR/LF character at the beginning",
            "description": "Werkzeug multipart data parser needs to find a boundary that may be between consecutive chunks. That's why parsing is based on looking for newline characters. Unfortunately, code looking for partial boundary in the buffer is written inefficiently, so if we upload a file that starts with CR or LF and then is followed by megabytes of data without these characters: all of these bytes are appended chunk by chunk into internal bytearray and lookup for boundary is performed on growing buffer.\n\nThis allows an attacker to cause a denial of service by sending crafted multipart data to an endpoint that will parse it. The amount of CPU time required can block worker processes from handling legitimate requests. The amount of RAM required can trigger an out of memory kill of the process. If many concurrent requests are sent continuously, this can exhaust or kill all available workers.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-hrfv-mqp8-q5rw",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-46136",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/werkzeug/security/advisories/GHSA-hrfv-mqp8-q5rw"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/b1916c0c083e0be1c9d887ee2f3d696922bfc5c1"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-46136"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/f3c803b3ade485a45f12b6d6617595350c0f03e2"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/werkzeug/PYSEC-2023-221.yaml"
                },
                {
                    "url": "https://github.com/pallets/werkzeug/commit/f2300208d5e2a5076cbbb4c2aad71096fd040ef9"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20231124-0008/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-hrfv-mqp8-q5rw"
                }
            ],
            "published_at": "2023-10-25T14:22:59Z",
            "updated_at": "2023-12-03T05:04:51Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "werkzeug"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 3.0.0, < 3.0.1",
                    "first_patched_version": {
                        "identifier": "3.0.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "werkzeug"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.3.8",
                    "first_patched_version": {
                        "identifier": "2.3.8"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:A/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
                "score": 5.7
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                },
                {
                    "cwe_id": "CWE-787",
                    "name": "Out-of-bounds Write"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "werkzeug"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.3.8",
            "first_patched_version": {
                "identifier": "2.3.8"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/272",
        "html_url": "https://github.com/fcas/igel/security/dependabot/272",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 271,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-8ghj-p4vj-mr35",
            "cve_id": "CVE-2023-44271",
            "summary": "Pillow Denial of Service vulnerability",
            "description": "An issue was discovered in Pillow before 10.0.0. It is a Denial of Service that uncontrollably allocates memory to process a given task, potentially causing a service to crash by having it run out of memory. This occurs for truetype in ImageFont when textlength in an ImageDraw instance operates on a long text argument.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-8ghj-p4vj-mr35",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-44271",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-44271"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/pull/7244"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2023-227.yaml"
                },
                {
                    "url": "https://devhub.checkmarx.com/cve-details/CVE-2023-44271"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4"
                },
                {
                    "url": "https://github.com/advisories/GHSA-8ghj-p4vj-mr35"
                }
            ],
            "published_at": "2023-11-03T06:36:30Z",
            "updated_at": "2024-03-22T15:25:43Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 0, < 10.0.0",
                    "first_patched_version": {
                        "identifier": "10.0.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-400",
                    "name": "Uncontrolled Resource Consumption"
                },
                {
                    "cwe_id": "CWE-770",
                    "name": "Allocation of Resources Without Limits or Throttling"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 0, < 10.0.0",
            "first_patched_version": {
                "identifier": "10.0.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/271",
        "html_url": "https://github.com/fcas/igel/security/dependabot/271",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 270,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-g4mx-q9vg-27p4",
            "cve_id": "CVE-2023-45803",
            "summary": "urllib3's request body not stripped after redirect from 303 status changes request method to GET",
            "description": "urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 303 \"See Other\" after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although the behavior of removing the request body is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers.\n\nFrom [RFC 9110 Section 9.3.1](https://www.rfc-editor.org/rfc/rfc9110.html#name-get):\n\n> A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported.\n\n## Affected usages\n\nBecause the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable.\n\nBoth of the following conditions must be true to be affected by this vulnerability:\n\n* If you're using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON)\n* The origin service is compromised and starts redirecting using 303 to a malicious peer or the redirected-to service becomes compromised.\n\n## Remediation\n\nYou can remediate this vulnerability with any of the following steps:\n\n* Upgrade to a patched version of urllib3 (v1.26.18 or v2.0.7)\n* Disable redirects for services that you aren't expecting to respond with redirects with `redirects=False`.\n* Disable automatic redirects with `redirects=False` and handle 303 redirects manually by stripping the HTTP request body.\n",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-g4mx-q9vg-27p4",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-45803",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/4e50fbc5db74e32cabd5ccc1ab81fc103adfe0b3"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/b594c5ceaca38e1ac215f916538fb128e3526a36"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/releases/tag/1.26.18"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/releases/tag/2.0.7"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-45803"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9"
                },
                {
                    "url": "https://www.rfc-editor.org/rfc/rfc9110.html#name-get"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2023-212.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-g4mx-q9vg-27p4"
                }
            ],
            "published_at": "2023-10-17T20:15:25Z",
            "updated_at": "2023-11-06T16:24:46Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.0.0, < 2.0.7",
                    "first_patched_version": {
                        "identifier": "2.0.7"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 0, < 1.26.18",
                    "first_patched_version": {
                        "identifier": "1.26.18"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
                "score": 4.2
            },
            "cwes": [
                {
                    "cwe_id": "CWE-200",
                    "name": "Exposure of Sensitive Information to an Unauthorized Actor"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 0, < 1.26.18",
            "first_patched_version": {
                "identifier": "1.26.18"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/270",
        "html_url": "https://github.com/fcas/igel/security/dependabot/270",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 269,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "GitPython"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-cwvm-v4w8-q58c",
            "cve_id": "CVE-2023-41040",
            "summary": "Blind local file inclusion",
            "description": "### Summary\n\nIn order to resolve some git references, GitPython reads files from the `.git` directory, in some places the name of the file being read is provided by the user, GitPython doesn't check if this file is located outside the `.git` directory. This allows an attacker to make GitPython read any file from the system.\n\n### Details\n\nThis vulnerability is present in\n\nhttps://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/refs/symbolic.py#L174-L175\n\nThat code joins the base directory with a user given string without checking if the final path is located outside the base directory.\n\nI was able to exploit it from three places, but there may be more code paths that lead to it:\n\nhttps://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/repo/base.py#L605\n\nhttps://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/repo/base.py#L620\n\nhttps://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/index/base.py#L1353\n\n### PoC\n\nRunning GitPython within any repo should work, here is an example with the GitPython repo.\n\n```python\nimport git\n\nr = git.Repo(\".\")\n\n# This will make GitPython read the README.md file from the root of the repo\nr.commit(\"../README.md\")\nr.tree(\"../README.md\")\nr.index.diff(\"../README.md\")\n\n# Reading /etc/random\n# WARNING: this will probably halt your system, run with caution\n# r.commit(\"../../../../../../../../../dev/random\")\n```\n\n### Impact\n\nI wasn't able to show the contents of the files (that's why \"blind\" local file inclusion), depending on how GitPython is being used, this can be used by an attacker for something _inoffensive_ as checking if a file exits, or cause a DoS by making GitPython read a big/infinite file (like `/dev/random` on Linux systems).\n\n### Possible solutions\n\nA solution would be to check that the final path isn't located outside the `repodir` path (maybe even after resolving symlinks). Maybe there could be other checks in place to make sure that the reference names are valid.\n\n---\n\n> [!NOTE]\n> This vulnerability was reported via email, and it was decided to publish it here and make it public, so the community is aware of it, and a fix can be provided.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-cwvm-v4w8-q58c",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-41040",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-cwvm-v4w8-q58c"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-41040"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/refs/symbolic.py#L174-L175"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/commit/74e55ee4544867e1bd976b7df5a45869ee397b0b"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/gitpython/PYSEC-2023-165.yaml"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/pull/1672"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/releases/tag/3.1.37"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00036.html"
                },
                {
                    "url": "https://github.com/advisories/GHSA-cwvm-v4w8-q58c"
                }
            ],
            "published_at": "2023-08-30T20:09:36Z",
            "updated_at": "2023-11-05T05:04:43Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "GitPython"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "<= 3.1.36",
                    "first_patched_version": {
                        "identifier": "3.1.37"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L",
                "score": 4.0
            },
            "cwes": [
                {
                    "cwe_id": "CWE-22",
                    "name": "Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "GitPython"
            },
            "severity": "medium",
            "vulnerable_version_range": "<= 3.1.36",
            "first_patched_version": {
                "identifier": "3.1.37"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/269",
        "html_url": "https://github.com/fcas/igel/security/dependabot/269",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 268,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-56pw-mpj4-fxww",
            "cve_id": null,
            "summary": "Bundled libwebp in Pillow vulnerable",
            "description": "Pillow versions before v10.0.1 bundled libwebp binaries in wheels that are vulnerable to CVE-2023-5129 (previously CVE-2023-4863). Pillow v10.0.1 upgrades the bundled libwebp binary to v1.3.2.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-56pw-mpj4-fxww",
                    "type": "GHSA"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-4863"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-5129"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2023-175.yaml"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst#1001-2023-09-15"
                },
                {
                    "url": "https://github.com/advisories/GHSA-56pw-mpj4-fxww"
                }
            ],
            "published_at": "2023-10-05T00:06:58Z",
            "updated_at": "2023-10-05T00:06:58Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 10.0.1",
                    "first_patched_version": {
                        "identifier": "10.0.1"
                    }
                }
            ],
            "cvss": {
                "vector_string": null,
                "score": 0.0
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "pillow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 10.0.1",
            "first_patched_version": {
                "identifier": "10.0.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/268",
        "html_url": "https://github.com/fcas/igel/security/dependabot/268",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 267,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-j7hp-h8jx-5ppr",
            "cve_id": "CVE-2023-4863",
            "summary": "libwebp: OOB write in BuildHuffmanTable",
            "description": "Heap buffer overflow in libwebp allow a remote attacker to perform an out of bounds memory write via a crafted HTML page. ",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-j7hp-h8jx-5ppr",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-4863",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-4863"
                },
                {
                    "url": "https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html"
                },
                {
                    "url": "https://crbug.com/1479274"
                },
                {
                    "url": "https://github.com/qnighy/libwebp-sys2-rs/pull/21"
                },
                {
                    "url": "https://github.com/qnighy/libwebp-sys2-rs/commit/4560c473a76ec8bd8c650f19ddf9d7a44f719f8b"
                },
                {
                    "url": "https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a"
                },
                {
                    "url": "https://bugzilla.suse.com/show_bug.cgi?id=1215231"
                },
                {
                    "url": "https://en.bandisoft.com/honeyview/history/"
                },
                {
                    "url": "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863"
                },
                {
                    "url": "https://news.ycombinator.com/item?id=37478403"
                },
                {
                    "url": "https://rustsec.org/advisories/RUSTSEC-2023-0060.html"
                },
                {
                    "url": "https://security-tracker.debian.org/tracker/CVE-2023-4863"
                },
                {
                    "url": "https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/"
                },
                {
                    "url": "https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/"
                },
                {
                    "url": "https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39823"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39825"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39826"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39827"
                },
                {
                    "url": "https://github.com/electron/electron/pull/39828"
                },
                {
                    "url": "https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/"
                },
                {
                    "url": "https://github.com/webmproject/libwebp/releases/tag/v1.3.2"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/"
                },
                {
                    "url": "https://rustsec.org/advisories/RUSTSEC-2023-0061.html"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202309-05"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5496"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5497"
                },
                {
                    "url": "https://www.debian.org/security/2023/dsa-5498"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/"
                },
                {
                    "url": "https://blog.isosceles.com/the-webp-0day/"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/21/4"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/1"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/3"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/4"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/5"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/6"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/7"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/22/8"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/26/1"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/26/7"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/28/1"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/28/2"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2023/09/28/4"
                },
                {
                    "url": "https://github.com/jaredforth/webp/pull/30"
                },
                {
                    "url": "https://github.com/python-pillow/Pillow/pull/7395"
                },
                {
                    "url": "https://github.com/jaredforth/webp/commit/9d4c56e63abecc777df71c702503c3eaabd7dcbc"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/"
                },
                {
                    "url": "https://pillow.readthedocs.io/en/stable/releasenotes/10.0.1.html#security"
                },
                {
                    "url": "https://security.netapp.com/advisory/ntap-20230929-0011/"
                },
                {
                    "url": "https://github.com/ImageMagick/ImageMagick/discussions/6664"
                },
                {
                    "url": "https://github.com/dlemstra/Magick.NET/releases/tag/13.3.0"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202401-10"
                },
                {
                    "url": "https://sethmlarson.dev/security-developer-in-residence-weekly-report-16"
                },
                {
                    "url": "https://www.bentley.com/advisories/be-2023-0001/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-j7hp-h8jx-5ppr"
                }
            ],
            "published_at": "2023-09-12T15:30:20Z",
            "updated_at": "2024-01-08T15:38:39Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "rust",
                        "name": "libwebp-sys2"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 0.1.8",
                    "first_patched_version": {
                        "identifier": "0.1.8"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rust",
                        "name": "libwebp-sys"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 0.9.3",
                    "first_patched_version": {
                        "identifier": "0.9.3"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 22.0.0, < 22.3.24",
                    "first_patched_version": {
                        "identifier": "22.3.24"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 24.0.0, < 24.8.3",
                    "first_patched_version": {
                        "identifier": "24.8.3"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 25.0.0, < 25.8.1",
                    "first_patched_version": {
                        "identifier": "25.8.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 26.0.0, < 26.2.1",
                    "first_patched_version": {
                        "identifier": "26.2.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "npm",
                        "name": "electron"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 27.0.0-beta.1, < 27.0.0-beta.2",
                    "first_patched_version": {
                        "identifier": "27.0.0-beta.2"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "SkiaSharp"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2.0.0, < 2.88.6",
                    "first_patched_version": {
                        "identifier": "2.88.6"
                    }
                },
                {
                    "package": {
                        "ecosystem": "go",
                        "name": "github.com/chai2010/webp"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 1.0.0",
                    "first_patched_version": null
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Pillow"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 10.0.1",
                    "first_patched_version": {
                        "identifier": "10.0.1"
                    }
                },
                {
                    "package": {
                        "ecosystem": "rust",
                        "name": "webp"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 0.2.6",
                    "first_patched_version": {
                        "identifier": "0.2.6"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q16-anycpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q16-hdri-anycpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q16-x64"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q8-anycpu"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q8-openmp-x64"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                },
                {
                    "package": {
                        "ecosystem": "nuget",
                        "name": "magick.net-q8-x64"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "< 13.3.0",
                    "first_patched_version": {
                        "identifier": "13.3.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "score": 8.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-787",
                    "name": "Out-of-bounds Write"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Pillow"
            },
            "severity": "high",
            "vulnerable_version_range": "< 10.0.1",
            "first_patched_version": {
                "identifier": "10.0.1"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/267",
        "html_url": "https://github.com/fcas/igel/security/dependabot/267",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 266,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-v845-jxx5-vc9f",
            "cve_id": "CVE-2023-43804",
            "summary": "`Cookie` HTTP header isn't stripped on cross-origin redirects",
            "description": "urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly.\n\nUsers **must** handle redirects themselves instead of relying on urllib3's automatic redirects to achieve safe processing of the `Cookie` header, thus we decided to strip the header by default in order to further protect users who aren't using the correct approach.\n\n## Affected usages\n\nWe believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited:\n\n* Using an affected version of urllib3 (patched in v1.26.17 and v2.0.6)\n* Using the `Cookie` header on requests, which is mostly typical for impersonating a browser.\n* Not disabling HTTP redirects\n* Either not using HTTPS or for the origin server to redirect to a malicious origin.\n\n## Remediation\n\n* Upgrading to at least urllib3 v1.26.17 or v2.0.6\n* Disabling HTTP redirects using `redirects=False` when sending requests.\n* Not using the `Cookie` header.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-v845-jxx5-vc9f",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-43804",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb"
                },
                {
                    "url": "https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-43804"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/urllib3/PYSEC-2023-192.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-v845-jxx5-vc9f"
                }
            ],
            "published_at": "2023-10-02T23:27:05Z",
            "updated_at": "2023-11-04T00:30:22Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.0.0, < 2.0.6",
                    "first_patched_version": {
                        "identifier": "2.0.6"
                    }
                },
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "urllib3"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 1.26.17",
                    "first_patched_version": {
                        "identifier": "1.26.17"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N",
                "score": 5.9
            },
            "cwes": [
                {
                    "cwe_id": "CWE-200",
                    "name": "Exposure of Sensitive Information to an Unauthorized Actor"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "urllib3"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 1.26.17",
            "first_patched_version": {
                "identifier": "1.26.17"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/266",
        "html_url": "https://github.com/fcas/igel/security/dependabot/266",
        "created_at": "2024-05-24T01:34:19Z",
        "updated_at": "2024-05-24T01:34:19Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 265,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-v8gr-m533-ghj9",
            "cve_id": null,
            "summary": "Vulnerable OpenSSL included in cryptography wheels",
            "description": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 2.5-41.0.3 are vulnerable to several security issues. More details about the vulnerabilities themselves can be found in https://www.openssl.org/news/secadv/20230908.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.",
            "severity": "low",
            "identifiers": [
                {
                    "value": "GHSA-v8gr-m533-ghj9",
                    "type": "GHSA"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pyca/cryptography/security/advisories/GHSA-v8gr-m533-ghj9"
                },
                {
                    "url": "https://github.com/pyca/cryptography/commit/fc11bce6930e591ce26a2317b31b9ce2b3e25512"
                },
                {
                    "url": "https://github.com/advisories/GHSA-v8gr-m533-ghj9"
                }
            ],
            "published_at": "2023-09-21T17:07:01Z",
            "updated_at": "2023-09-21T17:07:01Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "cryptography"
                    },
                    "severity": "low",
                    "vulnerable_version_range": ">= 2.5, < 41.0.4",
                    "first_patched_version": {
                        "identifier": "41.0.4"
                    }
                }
            ],
            "cvss": {
                "vector_string": null,
                "score": 0.0
            },
            "cwes": []
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "cryptography"
            },
            "severity": "low",
            "vulnerable_version_range": ">= 2.5, < 41.0.4",
            "first_patched_version": {
                "identifier": "41.0.4"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/265",
        "html_url": "https://github.com/fcas/igel/security/dependabot/265",
        "created_at": "2024-05-24T01:34:18Z",
        "updated_at": "2024-05-24T01:34:18Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 264,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "gitpython"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-wfm5-v35h-vwf4",
            "cve_id": "CVE-2023-40590",
            "summary": "GitPython untrusted search path on Windows systems leading to arbitrary code execution",
            "description": "### Summary\n\nWhen resolving a program, Python/Windows look for the current working directory, and after that the PATH environment (see big warning in https://docs.python.org/3/library/subprocess.html#popen-constructor). GitPython defaults to use the `git` command, if a user runs GitPython from a repo has a `git.exe` or `git` executable, that program will be run instead of the one in the user's `PATH`.\n\n### Details\n\nThis is more of a problem on how Python interacts with Windows systems, Linux and any other OS aren't affected by this. But probably people using GitPython usually run it from the CWD of a repo.\n\nThe execution of the `git` command happens in\n\nhttps://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/cmd.py#L277 \n\nhttps://github.com/gitpython-developers/GitPython/blob/1c8310d7cae144f74a671cbe17e51f63a830adbf/git/cmd.py#L983-L996\n\nAnd there are other commands executed that should probably be aware of this problem.\n\n### PoC\n\nOn a Windows system, create a `git.exe` or `git` executable in any directory, and import or run GitPython from that directory\n\n```\npython -c \"import git\"\n```\n\nThe git executable from the current directory will be run.\n\n### Impact\n\nAn attacker can trick a user to download a repository with a malicious `git` executable, if the user runs/imports GitPython from that directory, it allows the attacker to run any arbitrary commands.\n\n### Possible solutions\n \n- Default to an absolute path for the git program on Windows, like `C:\\\\Program Files\\\\Git\\\\cmd\\\\git.EXE` (default git path installation).\n- Require users to set the `GIT_PYTHON_GIT_EXECUTABLE` environment variable on Windows systems.\n- Make this problem prominent in the documentation and advise users to never run GitPython from an untrusted repo, or set the `GIT_PYTHON_GIT_EXECUTABLE` env var to an absolute path.\n- Resolve the executable manually by only looking into the `PATH` environment variable (suggested by @Byron)\n\n---\n\n> [!NOTE]\n> This vulnerability was reported via email, and it was decided to publish it here and make it public, so the community is aware of it, and a fix can be provided.",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-wfm5-v35h-vwf4",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-40590",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/gitpython-developers/GitPython/security/advisories/GHSA-wfm5-v35h-vwf4"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-40590"
                },
                {
                    "url": "https://docs.python.org/3/library/subprocess.html#popen-constructor"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/issues/1635"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/pull/1636"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/commit/8b75434e2c8a082cdeb4971cc6f0ee2bafec45bc"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/releases/tag/3.1.33"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/gitpython/PYSEC-2023-161.yaml"
                },
                {
                    "url": "https://github.com/advisories/GHSA-wfm5-v35h-vwf4"
                }
            ],
            "published_at": "2023-08-29T23:33:53Z",
            "updated_at": "2023-11-05T05:04:10Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "gitpython"
                    },
                    "severity": "high",
                    "vulnerable_version_range": "<= 3.1.32",
                    "first_patched_version": {
                        "identifier": "3.1.33"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "score": 7.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-426",
                    "name": "Untrusted Search Path"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "gitpython"
            },
            "severity": "high",
            "vulnerable_version_range": "<= 3.1.32",
            "first_patched_version": {
                "identifier": "3.1.33"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/264",
        "html_url": "https://github.com/fcas/igel/security/dependabot/264",
        "created_at": "2024-05-24T01:34:18Z",
        "updated_at": "2024-05-24T01:34:18Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 263,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "manifest_path": "poetry.lock",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-xqr8-7jwr-rhp7",
            "cve_id": "CVE-2023-37920",
            "summary": "Removal of e-Tugra root certificate",
            "description": "Certifi 2023.07.22 removes root certificates from \"e-Tugra\" from the root store. These are in the process of being removed from Mozilla's trust store.\n\n e-Tugra's root certificates are being removed pursuant to an investigation prompted by reporting of security issues in their systems. Conclusions of Mozilla's investigation can be found [here](https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A).",
            "severity": "high",
            "identifiers": [
                {
                    "value": "GHSA-xqr8-7jwr-rhp7",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-37920",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7"
                },
                {
                    "url": "https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-37920"
                },
                {
                    "url": "https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/certifi/PYSEC-2023-135.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-xqr8-7jwr-rhp7"
                }
            ],
            "published_at": "2023-07-25T14:43:53Z",
            "updated_at": "2023-11-09T05:05:47Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "certifi"
                    },
                    "severity": "high",
                    "vulnerable_version_range": ">= 2015.4.28, < 2023.7.22",
                    "first_patched_version": {
                        "identifier": "2023.7.22"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
                "score": 7.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-345",
                    "name": "Insufficient Verification of Data Authenticity"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "certifi"
            },
            "severity": "high",
            "vulnerable_version_range": ">= 2015.4.28, < 2023.7.22",
            "first_patched_version": {
                "identifier": "2023.7.22"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/263",
        "html_url": "https://github.com/fcas/igel/security/dependabot/263",
        "created_at": "2024-05-24T01:34:18Z",
        "updated_at": "2024-05-24T01:34:18Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 262,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "GitPython"
            },
            "manifest_path": "poetry.lock",
            "scope": "development"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-pr76-5cm5-w9cj",
            "cve_id": "CVE-2023-40267",
            "summary": "GitPython vulnerable to remote code execution due to insufficient sanitization of input arguments",
            "description": "GitPython before 3.1.32 does not block insecure non-multi options in `clone` and `clone_from`, making it vulnerable to Remote Code Execution (RCE) due to improper user input validation, which makes it possible to inject a maliciously crafted remote URL into the clone command. Exploiting this vulnerability is possible because the library makes external calls to git without sufficient sanitization of input arguments. NOTE: this issue exists because of an incomplete fix for CVE-2022-24439.",
            "severity": "critical",
            "identifiers": [
                {
                    "value": "GHSA-pr76-5cm5-w9cj",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-40267",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-40267"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/pull/1609"
                },
                {
                    "url": "https://github.com/gitpython-developers/GitPython/commit/ca965ecc81853bca7675261729143f54e5bf4cdd"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/gitpython/PYSEC-2023-137.yaml"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PF6AXUTC5BO7L2SBJMCVKJSPKWY52I5R/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AV5DV7GBLMOZT7U3Q4TDOJO5R6G3V6GH/"
                },
                {
                    "url": "https://github.com/advisories/GHSA-pr76-5cm5-w9cj"
                }
            ],
            "published_at": "2023-08-11T09:30:36Z",
            "updated_at": "2023-11-10T05:03:24Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "GitPython"
                    },
                    "severity": "critical",
                    "vulnerable_version_range": "< 3.1.32",
                    "first_patched_version": {
                        "identifier": "3.1.32"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
                "score": 9.8
            },
            "cwes": [
                {
                    "cwe_id": "CWE-78",
                    "name": "Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "GitPython"
            },
            "severity": "critical",
            "vulnerable_version_range": "< 3.1.32",
            "first_patched_version": {
                "identifier": "3.1.32"
            }
        },
        "url": "https://api.github.com/repos/fcas/igel/dependabot/alerts/262",
        "html_url": "https://github.com/fcas/igel/security/dependabot/262",
        "created_at": "2024-05-24T01:34:18Z",
        "updated_at": "2024-05-24T01:34:18Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    }
]